pyteal/ast/scratchvar_test.py

```diff
@@ -66,3 +67,27 @@ def test_scratchvar_load():
     actual = TealBlock.NormalizeBlocks(actual)
 
     assert actual == expected
+
+def test_scratchvar_assign_slot():
+    myScratch       = ScratchVar(TealType.uint64)
+    otherScratch    = ScratchVar(TealType.uint64, 1)
+    anotherScratch  = ScratchVar(TealType.uint64, 0)
+    lastScratch     = ScratchVar(TealType.uint64)
+    prog            = Seq([
+                        myScratch.store(Int(5)),      # Slot 2
+                        otherScratch.store(Int(0)),   # Slot 1
+                        anotherScratch.store(Int(7)), # Slot 0
+                        lastScratch.store(Int(9)),    # Slot 3
+                      ])
+
+    compileTeal(prog, mode=Mode.Signature, version=4) 
+
+def test_scratchvar_double_assign_invalid():
+    myvar    = ScratchVar(TealType.uint64, 10)
+    otherVar = ScratchVar(TealType.uint64, 10)
+    prog  = Seq([
+                myvar.store(Int(5)),
+                otherVar.store(Int(0))
+            ])
+    with pytest.raises(TealInternalError):
+        compileTeal(prog, mode=Mode.Signature, version=4) 
```

Comment:
These tests would actually fit in really well in the `pyteal/compiler/compiler_test.py` file, since they are testing parts of `compileTeal`. It would be great if you could compare the output of `compileTeal` to an expected source string too, which will help us guard against changing how slot assignments resolve in future versions.

For this file though, perhaps you could add version of `test_scratchvar_store` and `test_scratchvar_load` that use a user-defined slot id?

Reply:
I think that makes more sense -- I moved these tests to the `compile_test.py` file and added some unit tests regarding load/store here like you suggested.